staging/src/k8s.io/apiserver/pkg/registry/rest/resttest/resttest.go

```diff
@@ -749,16 +823,21 @@ func (t *Tester) testUpdateIgnoreClusterName(obj runtime.Object, createFn Create
 // =============================================================================
 // Deletion tests.
 
-func (t *Tester) testDeleteNoGraceful(obj runtime.Object, createFn CreateFunc, getFn GetFunc, isNotFoundFn IsErrorFunc) {
+func (t *Tester) testDeleteNoGraceful(obj runtime.Object, createFn CreateFunc, getFn GetFunc, isNotFoundFn IsErrorFunc, dryRun bool) {
 	ctx := t.TestContext()
 
 	foo := obj.DeepCopyObject()
 	t.setObjectMeta(foo, t.namer(1))
 	if err := createFn(ctx, foo); err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
+	defer t.delete(ctx, foo)
```

Comment:
I'm confused about this-- if not dryRun, it should have already been deleted at the end, and if dryRun it should not be removed at all? What am I missing?

Reply:
Added a comment.